### PR TITLE
releases: update changelog writing docs

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/11022

### CODE REVIEW

File: releases/changelog.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+**Those questions should be answered from the perspective of the user of the
+repository.** The user can be something else for different types of codebases:
+
+- For CLI program: a person or entity calling the binary.
+- For managed app: a person or entity installing the application.
+- For library: another codebase using the library.
+- For operator: a person or entity creating the CR consumed by this operator.
+
+## Examples of reasons to upgrade
```
Comment (MarcelMue) [inline releases/changelog.md:21]: I do not see the value in these examples - what do you want to explain with these reasons to upgrade?

Comment (kopiczko) [reply releases/changelog.md:21]: That they answer the question:

> - Is there a reason to upgrade to the new version?

And therefore should be included in the changelog. <- Maybe that's sentence that's missing or this is entirely stupid to add that section?

File: releases/changelog.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+- Changed the desired state of the system for the same CR. E.g. not public but
+  internal load-balancer is created for the same CR.
+
+## Examples of noise entries
+
+We should try to not mechanically write in changelog everything that is changed
+in the repository because not all changes are breaking and not all of them are
+the reason to upgrade for the end user.
+
+Here are some examples of not useful changelog entires creating unnecessary
```
Comment (oponder) [inline releases/changelog.md:56]:
```suggestion
Here are some examples of not useful changelog entries creating unnecessary
```

File: releases/changelog.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+documentation changes are not a reason to upgrade. They are also not released
+and non-breaking.
+
+- Rename internal variable from A to B.
+- Refactor internal package code.
+- Update architect orb version in CircleCI.
+
+Those are not a breaking changes. Code refactoring and way of building the code
+is not a reason for the end user to update.
+a reason for the end user to update.
```
Comment (oponder) [inline releases/changelog.md:71]:
```suggestion

```

File: releases/changelog.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+For operator:
+
+- Changed defaults for CR empty fields.
+- Changed the desired state of the system for the same CR. E.g. not public but
+  internal load-balancer is created for the same CR.
+
+## Examples of noise entries
+
+We should try to not mechanically write in changelog everything that is changed
```
Comment (rossf7) [inline releases/changelog.md:52]: I hadn't picked up the downsides of noise changelog entries. So IMO these docs help a lot.

But we added this to the PR checklist template which may be contributing.

- [ ] Update changelog in CHANGELOG.md.

Should it be this?

- [ ]  Update changelog in CHANGELOG.md (if needed).

Comment (kopiczko) [reply releases/changelog.md:52]: That's a good call. I also wanted to make it a link to this page like this:

> - [ ] [Update changelog in CHANGELOG.md (if needed)](https://github.com/giantswarm/fmt/blob/34f6186bca7fed497a4c3eb676edb79127bde7a1/releases/changelog.md).

Comment (rossf7) [reply releases/changelog.md:52]: Nice +1 on linking to these docs.

File: releases/changelog.md (hunk 2 above)
Comment (kopiczko) [reply releases/changelog.md:56]: I'm dumb. I didn't commit your suggestions 🤦

Comment (kopiczko) [reply releases/changelog.md:56]: https://github.com/giantswarm/fmt/pull/116

**Pull request merged**